src/hb-ot-shape-complex-use-machine.rl

```diff
@@ -157,15 +156,15 @@ main := |*
 static void
 find_syllables (hb_buffer_t *buffer)
 {
-  unsigned int p, pe, eof, ts, te, act;
+  machine_index_t p (buffer);
+  unsigned int pe, eof, ts, te, act;
```

Comment:
Are we should the other vars are never arithmatically-operated on?  I think that's correct.  Don't know how to ensure it though.

Reply:
Maybe use a type that has int cast operators but no arithmetic?